### ISSUE TITLE
Expression on same line of `if` condition are part of the true branch

### ISSUE THREAD

**Issue by giordano** [JuliaLang/julia#52014]
```julia
julia> if true println("hello")
           1
       end
hello
1

julia> Meta.@lower if true println("hello")
           1
       end
:($(Expr(:thunk, CodeInfo(
    @ none within `top-level scope`
1 ─     goto #3 if not true
    @ REPL[13]:1 within `top-level scope`
2 ─     println("hello")
│   @ REPL[13]:2 within `top-level scope`
└──     return 1
3 ─     return nothing
))))
```
The `println("hello")` expression becomes part of the body of the true branch.  This "feature" is even more nasty when you do by mistake something like
```julia
if condition1 && condition2 condition3 && condition4
```
`condition3 && condition4` aren't evaluated at all as part of the conditions of the `if`.  This should arguably a parser/syntax error.

**Comment by barucden** [comment]
Would it mean to also disallow one-liners in the form
```julia
if cond f() end
```
?

These one-liners are quite common, e.g.,:
https://github.com/JuliaLang/julia/blob/f2a5fb9fc493c859a86e7e9ab806c8634efbd865/stdlib/LinearAlgebra/test/generic.jl#L81
https://github.com/JuliaLang/julia/blob/f2a5fb9fc493c859a86e7e9ab806c8634efbd865/stdlib/LinearAlgebra/src/dense.jl#L1346
https://github.com/JuliaLang/julia/blob/f2a5fb9fc493c859a86e7e9ab806c8634efbd865/base/strings/search.jl#L189

**Comment by adienes** [comment]
this is for sure a footgun I have both shot myself with and complained about, but strictly speaking I'm not sure it's a bug. maybe a better job for a linter

**Comment by gbaraldi** [comment]
I guess we should've disallowed this a while ago, and told users to use the ternary, which is explicit

**Comment by Keno** [comment]
I'd be in favor of requiring a semicolon, but it's probably technically breaking

**Comment by JeffBezanson** [comment]
This was intentional (many years ago) but I agree it's highly debatable. Would be breaking to change now.

**Comment by jariji** [comment]
I use 
```jl
if p a end
```
but I'd rather use 

```jl
if p then a
```
from https://github.com/JuliaLang/julia/issues/16389.

**Comment by KristofferC** [comment]
`p && a` is short and you learn to just pattern match what it means pretty quickly.

**Comment by jariji** [comment]
Yeah, it's just more symbols with weird special meanings. And `if p then a` also supports `if p then a else b` (common in functional languages) rather than needing to overload `?` and `:` which isn't necessary either.
